integration_tests/integration_tests/tests/test_volume_anomalies.py

```diff
@@ -69,18 +71,100 @@ def test_volume_anomalies_with_where_parameter(
 
     params = DBT_TEST_ARGS
     test_result = dbt_project.test(
-        data, test_id, DBT_TEST_NAME, params, as_model=as_model
+        test_id, DBT_TEST_NAME, params, data=data, as_model=as_model
     )
     assert test_result["status"] == "fail"
 
     params = dict(DBT_TEST_ARGS, where="payback = 'karate'")
-    test_result = dbt_project.test(
-        data, test_id, DBT_TEST_NAME, params, as_model=as_model
-    )
+    test_result = dbt_project.test(test_id, DBT_TEST_NAME, params, as_model=as_model)
     assert test_result["status"] == "pass"
 
     params = dict(DBT_TEST_ARGS, where="payback = 'ka-razy'")
-    test_result = dbt_project.test(
-        data, test_id, DBT_TEST_NAME, params, as_model=as_model
+    test_result = dbt_project.test(test_id, DBT_TEST_NAME, params, as_model=as_model)
+    assert test_result["status"] == "fail"
+
+
+def test_volume_anomalies_with_time_buckets(test_id: str, dbt_project: DbtProject):
+    now = datetime.utcnow()
+    data = [
+        {TIMESTAMP_COLUMN: cur_date.strftime(DATE_FORMAT)}
+        for cur_date in generate_dates(base_date=now, period="hours", days_back=2)
+        if cur_date < now - timedelta(hours=1)
+    ]
+    # This is a bug in the dbt package. The test should pass, but it fails.
+    # test_result = dbt_project.test(test_id, DBT_TEST_NAME, DBT_TEST_ARGS, data=data)
+    # assert test_result["status"] == "pass"
+
+    test_args = {
+        **DBT_TEST_ARGS,
+        "time_bucket": {"period": "hour", "count": 1},
+        "days_back": 1,
+    }
+    test_result = dbt_project.test(test_id, DBT_TEST_NAME, test_args, data=data)
+    assert test_result["status"] == "fail"
+
+
+def test_volume_anomalies_with_direction_spike(test_id: str, dbt_project: DbtProject):
+    data = [
+        {TIMESTAMP_COLUMN: cur_date.strftime(DATE_FORMAT)}
+        for cur_date in generate_dates(base_date=date.today())
+        if cur_date < cur_date.today() - timedelta(days=1)
+        for _ in range(1 if cur_date < cur_date.today() - timedelta(days=1) else 2)
+    ]
```

Comment:
this list comprehension is too complicated (ignore, i cant delete for some reason)